staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -302,8 +302,7 @@ func NewReflectorWithOptions(lw ListerWatcher, expectedType interface{}, store R
 		// Using klog.TODO() here because switching to a caller-provided contextual logger
 		// would require an API change and updating all existing call sites.
 		klog.TODO().V(2).Info(
-			"The provided ListWatcher doesn't support WatchList semantics. The feature will be disabled. If you are using a custom client, check the documentation of watchlist.DoesClientNotSupportWatchListSemantics() method",
-			"listWatcherType", fmt.Sprintf("%T", lw),
+			"The client used to build this informer/reflector doesn't support WatchList semantics. The feature will be disabled. This is expected in unit tests but not in production. For details, see the documentation of watchlist.DoesClientNotSupportWatchListSemantics().",
```

Comment:
This message looks better.

Regarding `klog.TODO`: the effect is that in a `go test` invocation (no `-v`!) where only one test fails, the message will be shown once for each reflector in *all* tests, not just for the one test which failed. This is exactly the kind of log spam that I am trying to get rid of with contextual logging.

Is this message really worth an API change?

It's not an API break because `ReflectorOptions` can be extended to include a logger, but it's work.

Perhaps print this message only once per process? It's always the same message.